plugins/woocommerce/bin/lint-branch.sh

```diff
@@ -10,12 +10,21 @@
 
 baseBranch=${1:-"trunk"}
 
+# Lint changed php-files to match code style.
 changedFiles=$(git diff $(git merge-base HEAD $baseBranch) --relative --name-only --diff-filter=d -- '*.php')
-
-# Only complete this if changed files are detected.
-if [[ -z $changedFiles ]]; then
-    echo "No changed files detected."
-    exit 0
+if [[ -n $changedFiles ]]; then
+    composer exec phpcs-changed -- -s --git --git-base $baseBranch $changedFiles
 fi
 
-composer exec phpcs-changed -- -s --git --git-base $baseBranch $changedFiles
+# Lint new (excl. renamed) php-files to contain strict types directive.
+newFiles=$(git diff $(git merge-base HEAD $baseBranch) --relative --name-only --diff-filter=dmr -- '*.php')
+if [[ -n $newFiles ]]; then
+	passingFiles=$(find $newFiles -type f -exec grep -xl --regexp='declare(\s*strict_types\s*=\s*1\s*);' /dev/null {} +)
+	violatingFiles=$(grep -vxf <(printf "%s\n" $passingFiles | sort) <(printf "%s\n" $newFiles | sort))
+	if [[ -n violatingFiles ]]; then
```

Comment:
[This is failing for me](https://github.com/woocommerce/woocommerce/actions/runs/9759856799/job/26937578981?pr=48543#step:4:25), I guess because there's a newline character in an otherwise empty string?

This change seems to fix it for me locally.

```suggestion
	if [[ -n "$violatingFiles" ]]; then
```

Reply:
Thank you @opr! Fixed in https://github.com/woocommerce/woocommerce/pull/49015.